libc/unistd_uids.c

```diff
@@ -90,13 +90,15 @@ TEST(unistd_uids, setpuids_setsid)
 		child[1] = getpgrp();
 		child[2] = getsid(getpid());
 
-		siderr = setsid();
+		sidret = setsid();
 
 		child[3] = getpid();
 		child[4] = getpgrp();
 		child[5] = getsid(getpid());
+		/* sid return value == sid */
+		TEST_ASSERT_EQUAL_INT(sidret, child[5]);
 
-		_exit(siderr != 0);
+		_exit(0);
```

Comment:
Throwing unity errors not from main thread causes strange behaviors as the test may continue from the main thread.

Reply:
Yeah, in case of fail it would be handled incorrectly. Switched to volatile local variables.